src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java

```diff
@@ -0,0 +1,109 @@
+    private MockMvc mvc;
+
+    @Test
+    void should_return_200_when_ocr_form_validation_request_data_is_valid() throws Throwable {
+        String content = readResource("ocr-data/valid/valid-ocr-data.json");
+
+        mvc.perform(
+            post("/validate-ocr-data")
+                .header("ServiceAuthorization", "auth-header-value")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(content))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.status").value("SUCCESS"))
+            .andExpect(jsonPath("$.warnings", hasSize(0)))
+            .andExpect(jsonPath("$.errors", hasSize(0)));
+    }
+
+    @Test
+    void should_return_errors_when_ocr_data_contains_duplicate_fields() throws Throwable {
+        String content = readResource("ocr-data/invalid/duplicate-fields.json");
+
+        mvc.perform(
+            post("/validate-ocr-data")
+                .header("ServiceAuthorization", "auth-header-value")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(content))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.status").value("ERRORS"))
+            .andExpect(jsonPath("$.warnings", hasSize(0)))
+            .andExpect(jsonPath("$.errors", hasSize(1)))
+            .andExpect(jsonPath("$.errors[0]").value("Invalid OCR data. Duplicate keys exist: last_name"));
+    }
+
+    @Test
+    void should_return_errors_when_ocr_form_data_has_missing_mandatory_fields() throws Throwable {
+        String content = readResource("ocr-data/invalid/missing-mandatory-fields.json");
+
+        mvc.perform(
+            post("/validate-ocr-data")
+                .header("ServiceAuthorization", "auth-header-value")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(content))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.status").value("ERRORS"))
+            .andExpect(jsonPath("$.errors", hasSize(1)))
+            .andExpect(jsonPath("$.warnings", hasSize(0)))
+            .andExpect(jsonPath("$.errors[0]").value("last_name is missing"));
+    }
+
+    @Test
+    void should_return_warnings_when_ocr_form_data_has_missing_fields() throws Throwable {
+        String content = readResource("ocr-data/invalid/missing-optional-fields.json");
+
+        mvc.perform(
+            post("/validate-ocr-data")
+                .header("ServiceAuthorization", "auth-header-value")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(content))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.status").value("WARNINGS"))
+            .andExpect(jsonPath("$.errors", hasSize(0)))
+            .andExpect(jsonPath("$.warnings", hasSize(1)))
+            .andExpect(jsonPath("$.warnings[0]").value("date_of_birth is missing"));
+    }
+
+    @Test
+    void should_return_401_when_service_auth_header_is_missing() throws Throwable {
```

Comment:
need to add one more where service is not allowed to access the endpoint ?

Reply:
As discussed, this test scenario will not add any value as we have to mock the `AuthService` and it's already covered in unit test